test/locked_restart.jl

```diff
@@ -57,4 +57,21 @@ end
         r .-= V₁ * (V₁' * r)
         @test norm(r) ≤ ε
     end
+
+    # Test the orthonormality of Q
+    @test vecnorm(V[:,1:min]'*V[:,1:min] - I) < 1e-4
+    
+    # Test that H is upper triangular
+    @test vecnorm(triu(H)-H) < 1e-4
+
+    # for i = 1:min
+    #     for j = 1:min
+    #         if abs(H[i,j]) > 1e-9
+    #             @show (i,j)
+    #         end
+    #     end
+    # end
+
+    # Test the partial Schur decomposition relation AQ = QR
+    @test vecnorm(A*V[:,1:min] - V[:,1:min]*H[1:min,1:min]) < 1e-4
```

Comment:
Yeah, so this cannot hold because `H[min + 1, min]` was not zero. Basically, the first clusters converge quickly, but eigenvalues in the first cluster have a slower rate of convergence since they're closed tucked together.

And maybe if you do a gazillion restarts, the method might develop some numerical errors, spoiling convergence as well. Typically it V'V != I after enought restarts because of rounding errors. That's the pain of numerical linalg.